addons/hr_timesheet/controllers/portal.py

```diff
@@ -123,11 +122,16 @@ def get_timesheets():
             timesheets = Timesheet_sudo.search(domain, order=orderby, limit=_items_per_page, offset=pager['offset'])
             if field:
                 if groupby == 'date':
-                    time_data = Timesheet_sudo.read_group(domain, ['date', 'unit_amount:sum'], ['date:day'])
-                    mapped_time = dict([(datetime.strptime(m['date:day'], '%d %b %Y').date(), m['unit_amount']) for m in time_data])
-                    grouped_timesheets = [(Timesheet_sudo.concat(*g), mapped_time[k]) for k, g in groupbyelem(timesheets, itemgetter('date'))]
+                    raw_timesheets_group = Timesheet_sudo.read_group(
+                        domain, ["unit_amount:sum", "ids:array_agg(id)"], ["date:day"]
+                    )
+
+                    group_with_amount = [(timesheets & Timesheet_sudo.browse(group["ids"]), group["unit_amount"]) for group in raw_timesheets_group]
```

Comment:
Why do you need to check that the ids from the read_group are well part of timesheets? As both domains are the same, how would you get new ids in the read_group except in the context of a race condition where someone?
If that's the case, maybe just add that ids should be in the ones from timesheet.ids in the domain you pass to the read_goup, no?